File: C++/Hard/CompileError/split-array-largest-sum_4.cpp
class Solution {
public:
    int splitArray(vector<int>& nums, int k) {
        
        int n = nums.size();
        vector<int> sums(n + 1, 0);
        for (int i = 0; i < n; ++i) {
            sums[i + 1] = sums[i] + nums[i];
        }
        vector<vector<int>> dp(k + 1, vector<int>(n + 1, INT_MAX));
        dp[0][0] = 0;
        for (int i = int t = 0; t < j; ++t) {
                    dp[i][j] = min(dp[i][j], max(dp[i - 1][t], sum[j] - sum[t]));
                }
            }
        }
        return dp[k][n];
    }
};

File: Rust/Hard/Accepted/split-array-largest-sum_4.rs
impl Solution {
    pub fn split_array(nums: Vec<i32>, k: i32) -> i32 {
        
        let mut dp = vec![vec![std::i32::MAX; nums.len() + 1]; k as usize + 1];
        let mut sums = vec![0; nums.len() + 1];
        for i in 0..nums.len() {
            sums[i + 1] = sums[i] + nums[i];
        }
        dp[0][0] = 0;
        for i in 1..=k as usize {
            for j in 1..=nums.len() {
                for p in 0..j {
                    dp[i][j] = std::cmp::min(dp[i][j], std::cmp::max(dp[i - 1][p], sums[j] - sums[p]));
                }
            }
        }
        dp[k as usize][nums.len()]
    }
}